File: IslampurClotheEnventory/Controllers/PurchesController.cs
using IslampurClotheEnventory.Data.interfaces;
using IslampurClotheEnventory.Data.Models;
using IslampurClotheEnventory.Models;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using System;
using System.Security.Claims;

namespace IslampurClotheEnventory.Controllers
{
    [Authorize]
    public class PurchesController : Controller
    {
        private readonly IBasicServices _services;

        public PurchesController(IBasicServices services)
        {
            _services = services;
        }
        TosterResultViewModel result = new TosterResultViewModel();
        

        public IActionResult Index()
        {
            
            return View();
            
        }

        [HttpPost]
        public JsonResult Search([FromBody] Product product)
        {
            string userId = User.FindFirst(ClaimTypes.NameIdentifier).Value;
            return Json(_services.ProductSearch(product.ProductName,userId));
        }

        [HttpPost]
        public JsonResult Add([FromBody] PurchesViewModel purches)
        {
            if (ModelState.IsValid)
            {
                try
                {
                    Product product = new Product
                    {
                        ProductName = purches.ProductName,
                        ProductQuentity = purches.ProductQuentity,
                        ProductPurchesPrice = purches.ProductPurchesPrice,
                        ProductSalePrice = purches.ProductSalePrice,
                        IsStoed = purches.IsStoed,
                    };

                    if (_services.GetProductByName(purches.ProductName) == null)
                    {
                        _services.SetProduct(product);

                    }
                    else
                    {
                        _services.UpdateProductForPurches(product);
                    }

                    PurchesInfo purchesInfo = new PurchesInfo
                    {
                        PurchesPersonName = purches.PurchesPersonName,
                        PurchesPersonPhoneNumber = purches.PurchesPersonPhoneNumber,
                        PurchesPersonEmail = purches.PurchesPersonEmail,
                        PurchesDate = DateTime.Now,
                        PurchesPrice = purches.PurchesPrice,
                        PurchesOnCash = purches.PurchesOnCash,
                        PurchesOnDebt = purches.PurchesOnDebt,
                        ProductId = (_services.GetProductByName(purches.ProductName)).ProductId,
                        PurchesQuentity = purches.ProductQuentity

                    };
                    _services.SetPurchesInfo(purchesInfo);
                    result.IsSuccess = true;
                    result.Message = "Purches successfully complete";
                }
                catch (Exception ex)
                {

                    result.IsSuccess = false;
                    result.Message = ex.Message;
                }

            }

            return Json(result);
        }

        public JsonResult AllPurches()
        {
            return Json(_services.GetAllPurches());
        }

    }
}

File: IslampurClotheEnventory/wwwroot/js/angularScripts/Controllers/PurchesController.js
﻿
myApp.controller('Purches', function ($scope, PurchesServices) {

    toastr.options.positionClass = 'toast-bottom-right';
    toastr.options.newestOnTop = false;

    $scope.hidethis = true;

    $scope.search = function () {

        $scope.hidethis = false;
        var product = {
            productName: $scope.ProductName
        };

        PurchesServices.productsearch(product).then(function (response) {
            $scope.productList = response.data;
            console.log = response.data;

        }).catch(function onError(error) {
            console.log(error);
        });
    };


    $scope.FillTextBox = function (p) {
        $scope.ProductName = p.productName;

        $scope.isStoed = p.isStoed;
        $scope.hidethis = true;

    };

    $scope.cash = function () {
        $scope.OnDebt = $scope.TotalPrice - $scope.OnCash;
    };

    $scope.Add = function () {
        debugger
        var pur = {
            PurchesPersonName: $scope.Name,
            PurchesPersonPhoneNumber: $scope.PhoneNumber,
            PurchesPersonEmail: $scope.Email,
            PurchesPrice: $scope.TotalPrice,
            PurchesOnCash: $scope.OnCash,
            PurchesOnDebt: $scope.OnDebt,
            ProductName: $scope.ProductName,
            ProductQuentity: $scope.ProductQuentity,
            ProductPurchesPrice: $scope.ProductPurchesPrice,
            ProductSalePrice: $scope.ProductSalePrice,
            isStoed: $scope.isStoed
        };

        PurchesServices.addpurches(pur).then(function (response) {
            if (response.data.isSuccess) {
                toastr.success(response.data.message);
            }
            else {
                toastr.warning(response.data.message);
            }
            
        }).catch(function onError(error) {
            console.log(error);
        });

    };

    PurchesServices.getpurchesinfo().then(function (response) {
        $scope.purchesinfo = response.data;
        
    });

});